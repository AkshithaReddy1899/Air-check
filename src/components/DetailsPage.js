import React from 'react';
import { useLocation } from 'react-router-dom';

const DetailsPage = () => {
  const location = useLocation();
  return (
    <div>
      <h2>{location.state.item.city}</h2>
      <p>
        AQI:&nbsp;
        {location.state.item.aqi}
      </p>
      <p>
        <b>Carbon monoxide</b>
        :
        {location.state.item.co}
      </p>
      <p>
        <b>Ammonia</b>
        :
        {location.state.item.nh3}
      </p>
      <p>
        <b>Nitric oxide</b>
        :
        {location.state.item.no}
      </p>
      <p>
        <b>Nitrogen dioxide</b>
        :
        {location.state.item.no2}
      </p>
      <p>
        <b>Ozone</b>
        :
        {location.state.item.o3}
      </p>
      <p>
        <b>
          Fine particle
          <small>
            PM 2.5
          </small>
        </b>
        :
        {location.state.item.pm2_5}
      </p>
      <p>
        <b>
          Fine particle
          <small>
            PM10
          </small>
        </b>
        :
        {location.state.item.pm10}
      </p>
      <p>
        <b>Sulfur dioxide</b>
        :
        {location.state.item.so2}
      </p>
    </div>
  );
};

export default DetailsPage;
