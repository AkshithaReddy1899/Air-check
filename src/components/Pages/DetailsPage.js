import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import { getDetailsFromAPI } from '../../redux/details';
import '../../styles/details.css';

const DetailsPage = () => {
  const { state } = useParams();
  const dispatch = useDispatch();

  const data = useSelector((state) => state.details);
  const item = data.slice(-1);
  useEffect(() => {
    dispatch(getDetailsFromAPI(state));
  }, []);

  return (
    <>
      <div className="details">
        {
          item.map((obj) => (
            <li key={obj.id}>
              <div className="details-main">
                <h3>{(obj.name) ? obj.name : 'Not available'}</h3>
                <img src={obj.flag} alt="flag" className="flag" />
              </div>
              <small className="details-div">Country details</small>
              <p className="detail">
                <b>Official name</b>
                {
                  (obj.official) ? obj.official : 'Not available'
                }
              </p>
              <p className="detail odd">
                <b>continent</b>
                {
                  (obj.continent) ? obj.continent : 'Not available'
                }
              </p>
              <p className="detail">
                <b>Capital</b>
                {
                  (obj.cap) ? obj.cap : 'Not available'
                }
              </p>
              <p className="detail odd">
                <b>Region</b>
                {
                  (obj.region) ? obj.region : 'Not available'
                }
              </p>
              <p className="detail">
                <b>Sub Region</b>
                {
                  (obj.subregion) ? obj.subregion : 'Not available'
                }
              </p>
              <p className="detail odd">
                <b>Population</b>
                {
                  (obj.pop) ? obj.pop : 'Not available'
                }
              </p>
              <p className="detail">
                <b>Area</b>
                <span>
                  {
                    (obj.area) ? obj.area : 'Not available'
                  }
                  Km
                  <sup>2</sup>
                </span>
              </p>
            </li>
          ))
        }
      </div>
    </>
  );
};

export default DetailsPage;
