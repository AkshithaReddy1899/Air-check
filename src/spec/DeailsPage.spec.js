import React from 'react';
import { Provider } from 'react-redux';
import matches from './test-utils';
import store from '../redux/configStore';
import DetailsPage from '../components/Pages/DetailsPage';

describe('Details page', () => {
  it('DeailsPage matched snapshot', () => {
    matches(<Provider store={store}><DetailsPage /></Provider>);
  });
});
