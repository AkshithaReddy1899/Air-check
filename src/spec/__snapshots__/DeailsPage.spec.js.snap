// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details page DeailsPage matched snapshot 1`] = `
<div
  className="details"
>
  <a
    href="/"
    onClick={[Function]}
  >
    Back
  </a>
</div>
`;
